primer/src/Primer/Eval.hs

```diff
@@ -159,7 +159,7 @@ redexes tydefs globals =
 
 -- We hardcode a particular set of reduction options for the interactive evaluator
 reductionOpts :: RunRedexOptions
-reductionOpts = RunRedexOptions{}
+reductionOpts = RunRedexOptions{pushAndElide = False}
```

Comment:
In https://github.com/hackworthltd/primer/pull/736#discussion_r1273386023 we decided that the interactive eval should not do the pushAndElide bigger step. This has been implemented by hardcoding it (rather than making it an option), but this choice makes it slightly harder to test: previously the single-step eval testsuite `tests/Tests/Eval.hs` did `pushAndElide` and had the following tests of this functionality which now don't make sense
```
-- we should not elide @let x = ...@ here, as it is required for @let y@ which is not elided
unit_tryReduce_push_elide_1 :: Assertion
unit_tryReduce_push_elide_1 = do
  let (expr, i) =
        create $ let_ "x" (con0 cTrue) $ let_ "y" (lvar "x") $ lam "z" $ lvar "y"
  result <- runTryReduce tydefs mempty mempty (expr, i)
  case result of
    Right (expr', _) -> expr' ~= create' (lam "z" $ let_ "x" (con0 cTrue) $ let_ "y" (lvar "x") $ lvar "y")
    _ -> assertFailure $ show result

-- we should elide the outer @let x = ...@ in one branch, and the inner in the other
unit_tryReduce_push_elide_2 :: Assertion
unit_tryReduce_push_elide_2 = do
  let (expr, i) =
        create $ let_ "x" (con0 cTrue) $ let_ "y" (lvar "x") $ let_ "x" (con0 cFalse) $ app (lvar "x") (lvar "y")
  result <- runTryReduce tydefs mempty mempty (expr, i)
  case result of
    Right (expr', _) ->
      expr'
        ~= create'
          ( app
              (let_ "x" (con0 cFalse) $ lvar "x")
              (let_ "x" (con0 cTrue) $ let_ "y" (lvar "x") $ lvar "y")
          )
    _ -> assertFailure $ show result

-- we should not elide the outer @let x = ...@ here
unit_tryReduce_push_elide_3 :: Assertion
unit_tryReduce_push_elide_3 = do
  let (expr, i) =
        create $ let_ "x" (con0 cTrue) $ let_ "x" (lvar "x") $ lam "z" $ lvar "x"
  result <- runTryReduce tydefs mempty mempty (expr, i)
  case result of
    Right (expr', _) -> expr' ~= create' (lam "z" $ let_ "x" (con0 cTrue) $ let_ "x" (lvar "x") $ lvar "x")
    _ -> assertFailure $ show result
```
Additionally, the `unit_tryReduce_lettype_self_capture` needs changing if the step eval tests do push-and-elide.

Reply:
Can you leave a comment in the source where the hard-code is written, pointing back to the parent comment? (i.e., https://github.com/hackworthltd/primer/pull/736#discussion_r1293290757)